### PR TITLE
feat(frontend): add search contacts

### PR DESCRIPTION
**Pull request by CvetelinaAngelova**
# Motivation
As a user, I want to search my contacts by contact name so that I can quickly find and interact with specific people in my address book without manually scrolling through the entire list.

Screenshot 2025-05-15 at 19 17 09

# Acceptance Criteria
A search bar is visible when the address book has contacts.
As the user types in the search bar, the contact list is filtered in real-time.

 # Changes
Integrated a reusable InputSearch component for the search UI.
Bound the filter value to a reactive searchTerm variable in AddressBookStep.
Applied testId (ADDRESS_BOOK_SEARCH_CONTACT_INPUT) for test targeting.

 # Tests
Verified that the search input is rendered using ADDRESS_BOOK_SEARCH_CONTACT_INPUT.

### CODE REVIEW

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 1]
```diff
@@ -14,38 +18,56 @@
...
+					autofocus={true}
+					testId={ADDRESS_BOOK_SEARCH_CONTACT_INPUT}
+				/>
+			</div>
+			<div class="flex w-2/5 justify-end pt-1">
+				<Button
+					colorStyle="secondary-light"
+					on:click={() => addContact()}
+					testId={ADDRESS_BOOK_ADD_CONTACT_BUTTON}
+					styleClass="w-full h-full flex items-center justify-center gap-2 rounded-[12px]"
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:50]: is this `rounded-[12px]` for a new type of button style? I thought we were normalizing them. It starts to be a bit eclectic. If it is a style that it is going to be reused, may you please change it directly in `Button` component?

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 2]
```diff
@@ -14,38 +18,56 @@
...
-				</div>
-			{/each}
+			{#if filteredContacts.length > 0}
+				{#each filteredContacts as contact, index (index)}
+					<div class="flex items-center">
+						<div class="grow">
+							CONTACT: {contact.name} #addresses {contact.addresses.length}
+						</div>
+						<Button styleClass="flex-none" on:click={() => showContact(contact)}>Show</Button>
+					</div>
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:66]: i don't know if this part is temporary. If it is temporary, please add a TODO. If it is not temporary:

- aria label missing in button
- let's add a svelte transition
- localize i18n button label
- localize i18n for all the strings
- i would suggest to move it to another component, for better meneagability and re-usability. Something like ContactCard, or similar

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 3]
```diff
@@ -14,38 +18,56 @@
 		addContact: () => void;
 		showContact: (contact: Contact) => void;
 	}
+
 	let { contacts, addContact, showContact }: AddressBookStepProps = $props();
+
+	let searchTerm = $state('');
+
+	let filteredContacts = $derived(
+		contacts.filter((contact) => contact.name.toLowerCase().includes(searchTerm.toLowerCase()))
```
[the remark this answers is not in the document]
Comment (StefanBerger-DFINITY) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte:27]: I suggest to tokenize the text by empty spaces as separators and then apply an AND logic in a later improvement.

[the remark this answers is not in the document]
Comment (StefanBerger-DFINITY) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte:27]: Note: We are also not doing it here yet. (**Tether USD** is not found)
image

[the remark this answers is not in the document]
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte:27]: ah okok, then in another future improvemen

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 4]
```diff
@@ -14,38 +18,56 @@
...
+				{#each filteredContacts as contact, index (index)}
+					<div class="flex items-center">
+						<div class="grow">
+							CONTACT: {contact.name} #addresses {contact.addresses.length}
+						</div>
+						<Button styleClass="flex-none" on:click={() => showContact(contact)}>Show</Button>
+					</div>
+				{/each}
+			{:else}
+				<div class="text-gray-500">No contacts found</div>
```
[the remark this answers is not in the document]
Comment (FlavioInacta) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte:69]: Hm, we do not have a design for that. Maybe you can send a screenshot to @StefanBerger-DFINITY if it is ok like this?

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 1 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Add new Button Component - ButtonAdd.svelte 
according to figma the addButton should be with radius_small ( 12px ) and the main navigation buttons remind with radius 16px.
The new component will have default translation of "add" and IconPlus.

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Add some JS improvements which will pass the case ant ven and i want Antonio Ventilii.

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Changed to : 	<span class="text-gray-500">{$i18n.address_book.text.no_contact_found}</span>

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 2 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Add a TODO:

File: src/frontend/src/tests/lib/components/address-book/AddressBookStep.spec.ts [hunk 5]
```diff
@@ -130,4 +133,28 @@ describe('AddressBookStep', () => {
...
+		await fireEvent.input(input, { target: { value: 'Contact 1' } });
+
+		expect(queryByText('CONTACT: Test Contact 1 #addresses 1')).toBeInTheDocument();
+		expect(queryByText('CONTACT: Test Contact 2 #addresses 0')).not.toBeInTheDocument();
+
+		await fireEvent.input(input, { target: { value: '' } });
+
+		expect(queryByText('CONTACT: Test Contact 1 #addresses 1')).toBeInTheDocument();
+		expect(queryByText('CONTACT: Test Contact 2 #addresses 0')).toBeInTheDocument();
+	});
```
[the remark this answers is not in the document]
Comment (CvetelinaAngelova) [reply src/frontend/src/tests/lib/components/address-book/AddressBookStep.spec.ts:154]: Add some tests for hedge cases

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 6]
```diff
@@ -14,38 +18,60 @@
...
 		showContact: (contact: Contact) => void;
 	}
+
 	let { contacts, addContact, showContact }: AddressBookStepProps = $props();
+
+	let searchTerm = $state('');
+
+	let filteredContacts = $derived(
+		contacts.filter((contact) => {
+			const name = contact.name.toLowerCase();
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:28]:
```suggestion
		contacts.filter(({name: contactName}) => {
			const name = contactName.toLowerCase();
```

File: src/frontend/src/lib/components/ui/ButtonAdd.svelte [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	import { i18n } from '$lib/stores/i18n.store';
+
+	interface Props {
+		onclick: () => void;
+		testId?: string;
+		fullWidth?: boolean;
+		label?: string;
+	}
+
+	let { onclick, testId, fullWidth = true, label = $i18n.core.text.add }: Props = $props();
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/ui/ButtonAdd.svelte:13]: let's avoid default with derived; default to undefined and assign it as fallback in the logic

File: src/frontend/src/tests/lib/components/address-book/AddressBookStep.spec.ts [hunk 8]
```diff
@@ -1,32 +1,34 @@
...
-		},
-		{
-			id: 'contact-2',
-			name: 'Test Contact 2',
-			addresses: []
-		}
-	];
-
-	const mockAddContact = vi.fn();
-	const mockShowContact = vi.fn();
```
Comment (AntonioVentilii) [inline src/frontend/src/tests/lib/components/address-book/AddressBookStep.spec.ts:28]: why this change? does it not work if the mocks are inside the `describe` statement?

File: src/frontend/src/lib/i18n/en.json [hunk 9]
```diff
@@ -27,6 +27,7 @@
 			"no_results": "No results",
 			"paste": "Paste",
 			"to": "To",
+			"add": "add",
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/i18n/en.json:30]:
```suggestion
			"add": "Add",
```

File: src/frontend/src/lib/components/ui/ButtonAdd.svelte [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<script lang="ts">
+	import IconPlus from '$lib/components/icons/lucide/IconPlus.svelte';
+	import Button from '$lib/components/ui/Button.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+
+	interface Props {
+		onclick: () => void;
+		testId?: string;
+		fullWidth?: boolean;
+		label?: string;
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/ui/ButtonAdd.svelte:10]: we usually use children/slots to render the label, so that it can be reused with other things that are not only string.

Or, if you already know that it will always be that exact text, put it here directly, without passing it

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 11]
```diff
@@ -14,38 +18,60 @@
...
+								Address list item
+								https://github.com/dfinity/oisy-wallet/pull/6462
+							-->
+							CONTACT: {contact.name} #addresses {contact.addresses.length}
+						</div>
+						<Button styleClass="flex-none" on:click={() => showContact(contact)}>Show</Button>
+					</div>
+				{/each}
+			{:else}
+				<span class="text-gray-500">{$i18n.address_book.text.no_contact_found}</span>
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:73]: is there some theme-focused text class instead of `text-gray-500`?

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 12]
```diff
@@ -14,38 +18,60 @@
...
-					<Hr />
-				{/if}
-				<div class="flex items-center">
-					<div class="grow">CONTACT: {contact.name} #addresses {contact.addresses.length}</div>
-					<Button styleClass="flex-none" on:click={() => showContact(contact)}>Show</Button>
-				</div>
-			{/each}
+			{#if filteredContacts.length > 0}
+				{#each filteredContacts as contact, index (index)}
+					<div class="flex items-center">
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:61]: should we add the transition already here in this PR?

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 13]
```diff
@@ -14,38 +18,60 @@
...
+						<div class="grow">
+							<!-- TODO: Should be updated with Pull request #6462
+								Address list item
+								https://github.com/dfinity/oisy-wallet/pull/6462
+							-->
+							CONTACT: {contact.name} #addresses {contact.addresses.length}
+						</div>
+						<Button styleClass="flex-none" on:click={() => showContact(contact)}>Show</Button>
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:69]: this label will be localized too in another PR? if so, may you please add a TODO?

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: may you please inquiry if this is a new button style or if we can re-use one of the existing ones

File: src/frontend/src/lib/i18n/en.json (hunk 9 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/i18n/en.json:30]: Change it, but could also remove it - as we will remove the Add button component

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 1 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Yes I could. Get back to the standart button with rounded-[12px] for a styleClass. According to the design they want 12px border radius

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 12 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte:65]: This PR is only for the Search Contact functionality.

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte (hunk 13 above)
Comment (CvetelinaAngelova) [reply src/frontend/src/lib/components/address-book/AddressBookStep.svelte]: Todo is added. Should be changed with the List Item PR.

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 14]
```diff
@@ -14,38 +18,65 @@
...
+					showResetButton={notEmptyString(searchTerm)}
+					placeholder={$i18n.address_book.text.search_contact}
+					autofocus={true}
+					testId={ADDRESS_BOOK_SEARCH_CONTACT_INPUT}
+				/>
+			</div>
+			<div class="flex w-2/5 justify-end pt-[var(--padding)]">
+				<Button
+					colorStyle="secondary-light"
+					on:click={() => onAddContact()}
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:52]:
```suggestion
					on:click={onAddContact}
```

File: src/frontend/src/lib/components/address-book/AddressBookStep.svelte [hunk 15]
```diff
@@ -14,38 +18,65 @@
...
+					testId={ADDRESS_BOOK_ADD_CONTACT_BUTTON}
+					styleClass="rounded-[12px]"
+				>
+					<IconPlus></IconPlus>
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/address-book/AddressBookStep.svelte:56]:
```suggestion
					<IconPlus />
```

**Review by AntonioVentilii — APPROVED** [review]
LGTM tks!
